system/jlib/jtrace.cpp

```diff
@@ -765,14 +766,41 @@ class CTraceManager : implements ITraceManager, public CInterface
                         exporter = opentelemetry::exporter::trace::OStreamSpanExporterFactory::Create();
                         DBGLOG("Tracing to stdout/err...");
                     }
-                    else if (stricmp(exportType.str(), "OTLP")==0)
+                    else if (stricmp(exportType.str(), "OTLP")==0 || stricmp(exportType.str(), "OTLP-HTTP")==0)
+                    {
+                        opentelemetry::exporter::otlp::OtlpHttpExporterOptions trace_opts;
+                        if (exportConfig->hasProp("@url"))
+                        {
+                            StringBuffer url;
+                            exportConfig->getProp("@url", url);
+                            trace_opts.url = url.str();
+                        }
+                        else
+                            trace_opts.url = opentelemetry::exporter::otlp::GetOtlpDefaultHttpTracesEndpoint();
+
+                        if (exportConfig->hasProp("@timeOutSecs")) //not sure exactly what this value actually affects
+                        {
+                            trace_opts.timeout = std::chrono::seconds(exportConfig->getPropInt("@timeOutSecs"));
+                        }
+
+                        // Whether to print the status of the exporter in the console
+                        trace_opts.console_debug = exportConfig->getPropBool("@consoleDebug", false);
+
+                        exporter  = opentelemetry::exporter::otlp::OtlpHttpExporterFactory::Create(trace_opts);
+                        DBGLOG("Exporting traces via OTLP/HTTP to: (%s)", trace_opts.url.c_str());
+                    }
+                    else if (stricmp(exportType.str(), "OTLP-GRPC")==0)
                     {
                         namespace otlp = opentelemetry::exporter::otlp;
 
                         otlp::OtlpGrpcExporterOptions opts;
-                        StringBuffer endPoint;
-                        exportConfig->getProp("@endpoint", endPoint);
-                        opts.endpoint = endPoint.str();
+
+                        if (exportConfig->hasProp("@endpoint"))
+                        {
+                            StringBuffer endPoint;
+                            exportConfig->getProp("@endpoint", endPoint);
+                            opts.endpoint = endPoint.str();
+                        }
 
                         opts.use_ssl_credentials = exportConfig->getPropBool("@useSslCredentials", false);
```

Comment:
Worth discussing with business units whether they are using a secure protocol.  (I don't see any need.)  If so we will need some jiras to look at supporting that with secrets etc.